README.md

```diff
@@ -1,16 +1,1605 @@
+  update.task.meta: meta
+
+  update.can: crud/update
+  update.with: dns꞉example.com?TYPE=TXT
+  update.input: {value꞉"hello world"}
+  update.prf: UCAN
+  update.meta: {dev/tags"꞉ ["friends", "coffee"]}
+
+  update.task.do --> update.can
+  update.task.with --> update.with
+  update.task.input --> update.input
+  update.task.prf --> update.prf
+  update.task.meta --> update.meta
+
+}
+
+state send {
+  send.run: task
+  send.auth: authorization
+}
+
+state read {
+  read.run: task
+  read.auth: authorization
+}
+
+state update {
+  update.run: task
+  update.auth: authorization
+}
+
+
+state send.promise {
+  state send.promise.await <<choice>>
+}
+
+state read.promise {
+  state read.promise.await <<choice>>
+}
+
+state send.receipt {
+ send.receipt.sig: sig
+ send.receipt.ran: ran
+ send.receipt.out: out
+ send.receipt.fx: fx
+ send.receipt.meta: meta
+ send.receipt.iss: iss
+ send.receipt.prf: prf
+ --
+
+ state send.result <<choice>>
+ send.result --> send.result.ok: ok
+ send.result --> send.result.error: error
+
+ send.result.ok: success
+ send.result.error: { error }
+}
+
+
+send.run --> send.task
+send.auth --> auth
+
+read.run --> read.task
+read.auth --> auth
+
+update.run --> update.task
+update.auth --> auth
+
+scope --> send.task
+scope --> read.task
+scope --> update.task
+
+
+send.promise.await --> send: ok
+send.promise.await --> send: error
+send.promise.await --> send: *
+
+read.promise.await --> read.task: ok
+read.promise.await --> read.task: error
+read.promise.await --> read.task: *
+
+
+send.receipt.out --> send.result
+send.receipt.sig --> send.receipt.ran
+send.receipt.sig --> send.receipt.out
+send.receipt.sig --> send.receipt.fx
+send.receipt.sig --> send.receipt.meta
+send.receipt.sig --> send.receipt.iss
+send.receipt.sig --> send.receipt.prf
+
+
+send.receipt.ran --> send
+```
+
+### 2.2.1 Task
+
+A [Task] is like a deferred function application: a request to perform some action on a resource with specific input.
+
+### 2.2.2 Authorization
+
+An [Authorization] is a cryptographically signed proof permitting execution of referenced tasks. It allows invoker to authorize a group of tasks using one cryptographic signature.
+
+### 2.2.3 Invocation
+
+An [Invocation] is an invoker authorized instruction to an executor to run the [Task].
+
+### 2.2.4 Result
+
+A [Result] is the output of a [Task].
+
+### 2.2.5 Receipt
+
+A [Receipt] is a cryptographically signed description of the [Invocation] output.
+
+### 2.2.6 Promise
+
+A [promise] is a reference to an eventual [Receipt] of an [Invocation].
+
+## 2.3 IPLD Schema
+
+```ipldsch
+type Task<In> struct {
+  v       SemVer
+
+  with    URI
+  do      Ability
+
+  input   In (implicit {})
+  meta    {String : Any} (implicit {})
+  nnc     string (implicit "")
+
+  prf     [&UCAN] (implicit [])
+}
+
+type SemVer string
+type URI string
+
+type Authorization struct {
+  # Authorization is denoted by the set of links been authorized
+  scope   [&Any] (implicit [])
```

Comment:
I'm amendable to reconsidering, but idea was that you can lookup if CID is in the set. If we allow non links you may encounter them and it would not be obvious how to sort the content (which is desired for deterministic addressing).

All in all I think it's best to be conservative at the beginning and if we find need for more flexibility we can loosen things up as needed.

Reply:
Ah, okay that makes sense 👍 Also in this form we can more easily break it out into its own spec (signing over multiple things "in general").